native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -101,74 +102,94 @@ public static Schema getDecodedSchema(BString schemaBString) {
+                                                     Boolean validation) {
         BString fieldName = fieldObject.getObjectValue(INTERNAL_NODE).getStringValue(NAME_FIELD);
         ServiceType serviceType = (ServiceType) TypeUtils.getType(service);
         UnionType typeUnion = TypeCreator.createUnionType(PredefinedTypes.TYPE_STREAM, PredefinedTypes.TYPE_ERROR);
         for (ResourceMethodType resourceMethod : serviceType.getResourceMethods()) {
             if (SUBSCRIBE_ACCESSOR.equals(resourceMethod.getAccessor()) &&
                     fieldName.getValue().equals(resourceMethod.getResourcePath()[0])) {
-                ArgumentHandler argumentHandler = new ArgumentHandler(resourceMethod, context, fieldObject);
-                Object[] args = argumentHandler.getArguments();
-                ObjectType objectType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(service));
-                if (objectType.isIsolated() && objectType.isIsolated(resourceMethod.getName())) {
-                    env.getRuntime()
-                            .invokeMethodAsyncConcurrently(service, resourceMethod.getName(), null,
-                                                           null, executionCallback, null, typeUnion, args);
-                } else {
-                    env.getRuntime()
-                            .invokeMethodAsyncSequentially(service, resourceMethod.getName(), null,
-                                                           null, executionCallback, null, typeUnion, args);
+                ArgumentHandler argumentHandler =
+                        new ArgumentHandler(resourceMethod, context, fieldObject, responseGenerator, validation);
+                try {
+                    argumentHandler.validateInputConstraint(environment);
+                    Future subscriptionFutureResult = environment.markAsync();
+                    ExecutionCallback executionCallback = new ExecutionCallback(subscriptionFutureResult);
+                    Object[] args = argumentHandler.getArguments();
+                    ObjectType objectType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(service));
+                    if (objectType.isIsolated() && objectType.isIsolated(resourceMethod.getName())) {
+                        environment.getRuntime()
+                                .invokeMethodAsyncConcurrently(service, resourceMethod.getName(), null,
+                                        null, executionCallback, null, typeUnion, args);
+                    } else {
+                        environment.getRuntime()
+                                .invokeMethodAsyncSequentially(service, resourceMethod.getName(), null,
+                                        null, executionCallback, null, typeUnion, args);
+                    }
+                } catch (ConstraintValidationException e) {
+                    return null;
                 }
```

Comment:
Only `argumentHandler.validateInputConstraint()` method can throw an exception here, right? In that case, shall we only wrap that line in a try-catch block and move the other code outside?